lib/ansible/modules/web_infrastructure/ansible_tower/tower_group.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/python
+#coding: utf-8 -*-
+
+# (c) 2017, Wayne Witzel III <wayne@riotousliving.com>
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: tower_group
+version_added: "2.3"
+short_description: create, update, or destroy Ansible Tower group.
+description:
+    - Create, update, or destroy Ansible Tower groups. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - The name to use for the group.
+      required: True
+    description:
+      description:
+        - The description to use for the group.
+      required: False
+      default: null
+    inventory:
+      description:
+        - Inventory the group should be made a member of.
+      required: True
+    variables:
+      description:
+        - Variables to use for the group, use '@' for a file.
+      required: False
+      default: null
+    credential:
+      description:
+        - Credential to use for the group.
+      required: False
+      default: null
+    source:
+      description:
+        - The source to use for this group.
+      required: False
+      default: null,
+      choices: ["manual", "file", "ec2", "rax", "vmware", "gce", "azure", "azure_rm", "openstack", "satellite6" , "cloudforms", "custom"]
+    source_regions:
+      description:
+        - Regions for cloud provider.
+      required: False
+      default: null
+    source_vars:
+      description:
+        - Override variables from source with variables from this field.
+      required: False
+      default: null
+    instance_filters:
+      description:
+        - Comma-separated list of filter expressions for matching hosts.
```

Comment:
instead of 'comma separated list' just make 'list' , type=list will accept both types of list and always guarantee a list object to the module